### DISCUSSION TITLE
Why nesting a component inside a component w/ Controller leads to infinite loop?

### DISCUSSION THREAD

**Discussion by vladev** [react-hook-form/react-hook-form#2988] [Q&A] [answered]
Just see the code attached. 

https://codesandbox.io/s/react-hook-form-infinite-loop-swmii?file=/src/App.js

If I move the `CustomTextUnput` outside the component (i.e. as a top-level function) - everything works.

Error:
```
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
```

**Comment by bluebill1049** [comment, marked as answer]
i think because you shouldn't place inner-component insider component, this make component keep mount and unmount during render.

**Reply by blackmouse572** [reply]
Hi, I have same problem but I'm not fully understand what you are saying. Can you please explain more ?
